deltacat/compute/compactor_v2/deletes/delete_strategy_equality_delete.py

```diff
@@ -0,0 +1,165 @@
+                )
+                continue
+            curr_boolean_mask = pc.is_in(
+                table[delete_column_name],
+                value_set=delete_table[delete_column_name],
+            )
+            curr_boolean_mask = equality_predicate_operation(
+                prev_boolean_mask, curr_boolean_mask
+            )
+            prev_boolean_mask = curr_boolean_mask
+        number_of_rows_before_dropping = len(table)
+        logger.debug(
+            f"Number of table rows before dropping: {number_of_rows_before_dropping}."
+        )
+        table = table.filter(pc.invert(curr_boolean_mask))
+        number_of_rows_after_dropping = len(table)
+        logger.debug(
+            f"Number of table rows after dropping: {number_of_rows_after_dropping}."
+        )
+        dropped_rows = number_of_rows_before_dropping - number_of_rows_after_dropping
+        return table, dropped_rows
+
+    def apply_deletes(
+        self,
+        table: Optional[pa.Table],
+        delete_file_envelope: DeleteFileEnvelope,
+        *args,
+        **kwargs,
+    ) -> Tuple[pa.Table, int]:
+        """
+        Apply delete operations on the given table using the provided delete file envelope.
+
+        Args:
+            table (Optional[pa.Table]): The pyarrow table to apply deletes on.
+            delete_file_envelope (DeleteFileEnvelope): The delete file envelope containing delete parameters.
+
+        Returns:
+            Tuple[pa.Table, int]: A tuple containing the updated Arrow table after applying deletes,
+                and the number of rows deleted.
+        """
+        if not table or not delete_file_envelope.table:
+            logger.debug(
+                f"No table passed or no delete file envelope delete table found. DeleteFileEnvelope: {delete_file_envelope}"
+            )
+            return table, 0
+        delete_columns = delete_file_envelope.delete_columns
+        delete_table = delete_file_envelope.table
+        table, number_of_rows_dropped = self._drop_rows(
+            table, delete_table, delete_columns
+        )
+        return table, number_of_rows_dropped
+
+    def apply_many_deletes(
+        self,
+        table: Optional[pa.Table],
+        delete_file_envelopes: List[DeleteFileEnvelope],
+        *args,
+        **kwargs,
+    ) -> Tuple[pa.Table, int]:
+        """
+        Apply delete operations on the given table using all provided delete file envelopes.
+
+        Args:
+            table (Optional[pa.Table]): The pyarrow table to apply deletes on.
+            delete_file_envelopes (List[DeleteFileEnvelope]): A list of delete file envelopes containing delete parameters.
+
+        Returns:
+            Tuple[pa.Table, int]: A tuple containing the updated pyarrow table after applying all deletes,
+                and the total number of rows deleted.
+        """
+        # looking up table references are lighter than the actual table
+        if not table or any(
```

Comment:
```suggestion
        if not table or all(
```

Reply:
Which test case fails on this bug?

Reply:
Is table_reference ever be None? You may want to add validation instead of silently failing.

Reply:
This test fails - https://github.com/ray-project/deltacat/blob/76ee0b06a2896cdeccf25454eb146fb324b807f2/deltacat/tests/compute/compactor_v2/deletes/test_delete_strategy_equality_delete.py#L302

Reply:
All delete files must not have reference to not drop deletes right?

Reply:
This assertion has been removed since we are checking now while iterating over the delete file envelopes

```
    def apply_many_deletes(
        self,
        table: Optional[pa.Table],
        delete_file_envelopes: List[DeleteFileEnvelope],
        *args,
        **kwargs,
    ) -> Tuple[pa.Table, int]:
        """
        Apply delete operations on the given table using all provided delete file envelopes.

        Args:
            table (Optional[pa.Table]): The pyarrow table to apply deletes on.
            delete_file_envelopes (List[DeleteFileEnvelope]): A list of delete file envelopes containing delete parameters.

        Returns:
            Tuple[pa.Table, int]: A tuple containing the updated pyarrow table after applying all deletes,
                and the total number of rows deleted.
        """
        # looking up table references are lighter than the actual table
        if not table:
            logger.debug(
                "No table passed or delete file envelopes are missing a delete table."
            )
            return table, 0
        total_dropped_rows = 0
        for delete_file_envelope in delete_file_envelopes:
            if delete_file_envelope.table_reference is None:
                continue
            table, number_of_rows_dropped = self.apply_deletes(
                table, delete_file_envelope
            )
            total_dropped_rows += number_of_rows_dropped
        return table, total_dropped_rows
```